ql/src/java/org/apache/hadoop/hive/ql/exec/ReplTxnTask.java

```diff
@@ -85,7 +85,7 @@ public int execute(DriverContext driverContext) {
     }
 
     try {
-      HiveTxnManager txnManager = driverContext.getCtx().getHiveTxnManager();
+      HiveTxnManager txnManager = driverContext.getReplTxnManager(conf);
```

Comment:
Noticed in several places, SessionState.get().getTxnMgr() is used to get txnmgr on which performed some operations. For example, addWriteNotificationLog, Hive.getTable etc.
I think, those operations would be pointing to source metastore. Might go wrong.

Reply:
yes ..the changes looks risky ..